skillserver/skillserver.go

```diff
@@ -58,12 +59,26 @@ func SetRootPrefix(prefix string) {
 	RootPrefix = prefix
 }
 
-func RunSSL(apps map[string]interface{}, port, cert, key string) error {
-	router := mux.NewRouter()
-	Init(apps, router)
-
-	err := http.ListenAndServeTLS(port, cert, key, router)
-	return err
+func RunSSL(apps map[string]interface{}, port, cert, key string) {
+  router := mux.NewRouter()
+  Init(apps, router)
+
+  cfg := &tls.Config{
+      MinVersion:               tls.VersionTLS12,
+      CurvePreferences:         []tls.CurveID{tls.CurveP521, tls.CurveP384, tls.CurveP256},
+      PreferServerCipherSuites: true,
+      CipherSuites: []uint16{
+				tls.TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256,
```

Comment:
Any reason not to inline this with `uint16{` above?

Reply:
This one might change over time, I think I will put a comment here stating one needs to try the possibilities listed in crypto/tls here for the one which works